addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/core/LxWsClient.java

```diff
@@ -0,0 +1,765 @@
+                if (trace.length() > 100) {
+                    trace = msg.substring(0, 100);
+                }
+                logger.trace("[{}] received message in state {}: {}", debugId, state.toString(), trace);
+            }
+            synchronized (state) {
+                try {
+                    LxJsonResponse resp;
+                    switch (state) {
+                        case IDLE:
+                        case CONNECTING:
+                            logger.debug("[{}] Unexpected message received by websocket in state {}", debugId,
+                                    state.toString());
+                            break;
+                        case CONNECTED:
+                            // expecting a key to hash credentials
+                            resp = new Gson().fromJson(msg, LxJsonResponse.class);
+                            if (resp.LL.Code == 420) {
+                                notifyAndClose(OfflineReason.AUTHENTICATION_TIMEOUT,
+                                        "Timeout on authentication procedure, response : " + resp.LL.value);
+                            } else if (resp.LL.control.equals(CMD_GET_KEY) && resp.LL.Code == 200) {
+                                String credentials = hashCredentials(resp.LL.value);
+                                if (credentials != null) {
+                                    sendString(CMD_AUTHENTICATE + credentials);
+                                    setClientState(ClientState.AUTHENTICATING);
+                                    startResponseTimeout();
+                                } else {
+                                    notifyAndClose(OfflineReason.INTERNAL_ERROR, "Error creating credentials");
+                                }
+                            }
+                            break;
+                        case AUTHENTICATING:
+                            resp = new Gson().fromJson(msg, LxJsonResponse.class);
+                            if (resp.LL.Code == 401) {
+                                notifyAndClose(OfflineReason.UNAUTHORIZED, "Websocket credentials unauthorized.");
+                            } else if (resp.LL.Code == 420) {
+                                notifyAndClose(OfflineReason.AUTHENTICATION_TIMEOUT,
+                                        "Timeout on authentication procedure, response : " + resp.LL.value);
+                            } else if (resp.LL.Code == 200) {
+                                logger.debug("[{}] Websocket authentication successfull.", debugId);
+                                sendString(CMD_GET_APP_CONFIG);
+                                setClientState(ClientState.UPDATING_CONFIGURATION);
+                                startResponseTimeout();
+                            }
+                            break;
+                        case UPDATING_CONFIGURATION:
+                            LxJsonApp3 config = new Gson().fromJson(msg, LxJsonApp3.class);
```

Comment:
Should you really create a new `Gson` everytime, I believe you can re-use it please check.